rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java

```diff
@@ -90,6 +91,7 @@
   // default values
   private static final int DEFAULT_BLOCK_SIZE = 1 * 1024 * 1024; // 1MB
   private static final int DEFAULT_SERVER_CONNECT_TIMEOUT = 1000; // ms
+  private static final int DEFAULT_SERVER_SOCKET_TIMEOUT = 3000; // ms
```

Comment:
I would suggest to make the default value to 6 sec as some of the initial S3 calls are taking 2-3 sec. So to be in the safe side, lets make it 6sec